shoplist/src/main/java/com/netguru/codereview/ui/MainViewModel.kt

```diff
@@ -0,0 +1,43 @@
+package com.netguru.codereview.ui
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.netguru.codereview.network.ShopListApiMock
+import com.netguru.codereview.network.ShopListRepository
+import com.netguru.codereview.network.model.ShopListItemResponse
+import com.netguru.codereview.network.model.ShopListResponse
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+class MainViewModel : ViewModel() {
+
+    private val shopListRepository = ShopListRepository(ShopListApiMock())
```

Comment:
```suggestion
class MainViewModel @Inject constructor(private val shopListRepository : ShopListRepository)
```

it is not good to initialize a new instance every time creation for the ShopListRepository, so we can inject an instance in the constructor